software/lazydogpile.livecodescript

```diff
@@ -0,0 +1,954 @@
+         put fitImageToConstraint(tQuery) into tGimme
+         set the width of image "figureHolder" stack "NodeDetails" to tGimme["width"]
+         set the height of image "figureHolder" stack "NodeDetails" to tGimme["height"]
+         set the loc of image "figureHolder" stack "NodeDetails" to tGimme["center"]
+      else if tDeets[pLine]["new"] is not empty then
+         -- this means there's an image file in node details that hasn't been committed to the graph yet
+         -- I guess we don't really have to to anything special for this
+         -- the image is already set up and the next part will make the right things visible
+      end if -- [figure] is not [empty]
+      set the visible of field "editData" stack "NodeDetails" to false
+      set the visible of field "tagLabel" stack "NodeDetails" to false
+      set the visible of image "figureHolder" stack "NodeDetails" to true
+      set the visible of button "loadImgButt" stack "NodeDetails" to true
+      set the visible of button "clearImgButt" stack "NodeDetails" to false 
+      set the visible of button "expandImgButt" stack "NodeDetails" to false 
+      set the visible of button "setCaptureButt" stack "NodeDetails" to true
+      set the visible of button "snapCaptureButt" stack "NodeDetails" to true
+      if tDeets[pLine]["figure"] is not "[empty]" or (tDeets[pLine]["new"] is not empty and tDeets[pLine]["new"] is not "hide") then
+         -- allow clear if 1) [figure] already has a file or 2) a new file has been loaded in [new]
+         -- don't allow clear if "hide" is in [new], which is trying to get rid of what's in [figure]
+         set the visible of button "clearImgButt" stack "NodeDetails" to true 
+         set the visible of button "expandImgButt" stack "NodeDetails" to true 
+      end if
+   else if tLabel is among the words "title memo weight" then
+      set the visible of field "editData" stack "NodeDetails" to true
+      set the visible of field "tagLabel" stack "NodeDetails" to true
+      set the visible of image "figureHolder" stack "NodeDetails" to false
+      set the visible of button "loadImgButt" stack "NodeDetails" to false
+      set the visible of button "clearImgButt" stack "NodeDetails" to false 
+      set the visible of button "expandImgButt" stack "NodeDetails" to false 
+      set the visible of button "setCaptureButt" stack "NodeDetails" to false
+      set the visible of button "snapCaptureButt" stack "NodeDetails" to false
+      put tLabel into field "tagLabel" stack "NodeDetails"
+      if tDeets[pLine]["new"] is not empty and tDeets[pLine]["new"] is not "hide" then
+         put tDeets[pLine]["new"] into field "editData" stack "NodeDetails"
+      else 
+         put tDeets[pLine][tLabel] into field "editData" stack "NodeDetails"
+      end if
+      select the text of field "editData" stack "NodeDetails"
+   end if -- if label is divider
+   set the cNodeDetails of stack "NodeDetails" to tDeets
```

Comment:
@matthewmaier I generally use `u` for custom properties rather than `c` following the [Fourth World - Scripting Style Guide](http://fourthworld.com/embassy/articles/scriptstyle.html#Variables)